tests/backup/backup_share_test.go

```diff
@@ -4215,3 +4215,400 @@ var _ = Describe("{SwapShareBackup}", func() {
+				log.FailOnError(err, "failed to delete backup %s uid", backupName)
+			}
+		})
+		Step(fmt.Sprintf("Take successful backup of applications from the user %s", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking successful backup of applications from the user %s", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			var mu sync.RWMutex
+			createBackup := func(backupName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+				for backupLocationUID, backupLocationName := range userBackupLocationMap {
+					err := CreateBackupWithValidation(nonAdminCtx, backupName, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup, make(map[string]string), orgID, userSourceClusterUID, "", "", "", "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation and validation of backup [%s] of namespace (scheduled Context) [%s]", backupName, namespace))
+					break
+				}
+				mu.Lock()
+				defer mu.Unlock()
+				userSuccessfulBackupMap[backupName] = namespace
+			}
+			for _, namespace := range appNamespaces {
+				for i := 0; i < numberOfBackups; i++ {
+					backupName := fmt.Sprintf("%s-%s-%d-%v", BackupNamePrefix, namespace, i, time.Now().Unix())
+					wg.Add(1)
+					go createBackup(backupName, namespace)
+				}
+			}
+			wg.Wait()
+			log.Infof("The list of successful user backups taken are: %v", userSuccessfulBackupMap)
+		})
+		Step(fmt.Sprintf("Take in progress restore of backups from the user %s", infraAdminUser), func() {
+			log.InfoD(fmt.Sprintf("Taking in progress restore of backups from the user %s", infraAdminUser))
+			nonAdminCtx, err := backup.GetNonAdminCtx(infraAdminUser, commonPassword)
+			log.FailOnError(err, "failed to fetch user %s ctx", infraAdminUser)
+			var wg sync.WaitGroup
+			var mu sync.RWMutex
+			createRestore := func(backupName string, restoreName string, namespace string) {
+				defer GinkgoRecover()
+				defer wg.Done()
+				namespaceMapping := map[string]string{namespace: fmt.Sprintf("r-%s-%v", namespace, time.Now().Unix())}
+				_, err = CreateRestoreWithoutCheck(restoreName, backupName, namespaceMapping, destinationClusterName, orgID, nonAdminCtx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying creation of restore %s of backup %s", restoreName, backupName))
+				mu.Lock()
+				defer mu.Unlock()
+				userInProgressRestoreNames = append(userInProgressRestoreNames, restoreName)
+			}
+			for backupName, namespace := range userSuccessfulBackupMap {
+				wg.Add(1)
+				restoreName := fmt.Sprintf("%s-%s", restoreNamePrefix, backupName)
+				go createRestore(backupName, restoreName, namespace)
+			}
+			wg.Wait()
+			log.Infof("The list of in progress user restores taken are: %v", userInProgressRestoreNames)
+		})
```

Comment:
I might be missing a step to verify restore by non-admin user is listed from admin pages while restore is in inprogress state. @ak-px

Reply:
@kphalgun-px , owner id for restore is not supported in enumerate options, you can drop listing for restore deletion steps

Reply:
Done. Used GetAllRestoreNamesByOwnerID in topic/krishna-px/PA-1528-v7.